python/Ganga/Utility/Setup.py

```diff
@@ -84,17 +84,41 @@ def prependPath(self, name, var):
                     pp = ''
                 os.environ[var] = path + pp
 
-    def setSysPath(self, name):
+    def setSysPath(self, name, set_python_path=True):
         """ Update the sys.path variable for a given package by adding a given package to the PYTHONPATH.
         It assumes that the first item in PYTHONPATH is Ganga, so the package path is added as a second item.
+        If set_python_path is True, it will update the PYTHONPATH as well
         """
         import sys
-        import os.path
+        import os
 
         if getExternalHome():
-            path = self.getPackagePath2(name, 'syspath')
-            if path:
-                sys.path.insert(1, path)
+
+            # get the path(s)
+            paths = self.getPackagePath2(name, 'syspath')
+            if not paths:
+                return True
+
+            # loop over the sys paths to be added
+            for path in reversed(paths.split(':')):
+                if path not in sys.path:
+
+                    # add to sys path
+                    sys.path.insert(1, path)
```

Comment:
Basically, because it assumes that Ganga is the first thing in the the system path (see the comment). Whether or not that's a safe assumption and where it's set is up for debate but it seems correct.